packages/react/src/table/TableScroller.tsx

```diff
@@ -0,0 +1,336 @@
+				mousePos.current.y = pageY;
+			}
+		},
+		[isDraggingThumb, thumbWidthRatio]
+	);
+
+	const handleThumbRelease = useCallback(() => {
+		if (isDraggingThumb) {
+			setIsDraggingThumb(false);
+		}
+	}, [isDraggingThumb]);
+
+	useEffect(() => {
+		if (isDraggingThumb) {
+			document.addEventListener('mousemove', handleThumbMove);
+			document.addEventListener('mouseup', handleThumbRelease);
+			document.addEventListener('touchmove', handleThumbMove);
+			document.addEventListener('touchend', handleThumbRelease);
+		} else {
+			document.removeEventListener('mousemove', handleThumbMove);
+			document.removeEventListener('mouseup', handleThumbRelease);
+			document.removeEventListener('touchmove', handleThumbMove);
+			document.removeEventListener('touchend', handleThumbRelease);
+		}
+
+		return () => {
+			document.removeEventListener('mousemove', handleThumbMove);
+			document.removeEventListener('mouseup', handleThumbRelease);
+			document.removeEventListener('touchmove', handleThumbMove);
+			document.removeEventListener('touchend', handleThumbRelease);
+		};
+	}, [handleThumbMove, handleThumbRelease, isDraggingThumb]);
+
+	const handleButtonClick = (direction: 'left' | 'right') => {
+		// Windows and keyboard left/right generally moves 40px, but it can change based on some ratio. Let's just keep it simple for now
+		const scrollAmount = direction === 'left' ? -40 : 40;
+
+		if (scrollerRef.current) {
+			scrollerRef.current.scrollLeft += scrollAmount;
+		}
+
+		if (buttonIntervalId) {
+			clearInterval(buttonIntervalId);
+			setButtonIntervalId(null);
+		}
+	};
+
+	const handleButtonPress = (
+		event: React.MouseEvent | React.TouchEvent,
+		direction: 'left' | 'right'
+	) => {
+		// Windows and keyboard left/right generally moves 40px, but it can change based on some ratio. Let's just keep it simple for now
+		const scrollAmount = direction === 'left' ? -40 : 40;
+
+		const intervalId = window.setInterval(() => {
+			if (
+				scrollerRef.current &&
+				(event.type === 'touchstart' ||
+					(event.type === 'mousedown' &&
+						'button' in event &&
+						event.button === 0))
+			) {
+				scrollerRef.current.scrollLeft += scrollAmount;
+			}
+		}, 100);
+
+		setButtonIntervalId(intervalId);
+	};
+
+	const handleButtonRelease = () => {
+		if (buttonIntervalId) {
+			clearInterval(buttonIntervalId);
+			setButtonIntervalId(null);
+		}
+	};
+
+	const handleTrackClick = (event: React.MouseEvent) => {
+		if (!scrollerRef?.current || !thumbRef?.current) {
+			return;
+		}
+
+		const thumbDimensions = thumbRef.current.getBoundingClientRect();
+
+		if (event.pageX > thumbDimensions.x + thumbDimensions.width) {
+			scrollerRef.current.scrollLeft += thumbDimensions.width * 0.95;
+		} else if (event.pageX < thumbDimensions.x) {
+			scrollerRef.current.scrollLeft -= thumbDimensions.width * 0.95;
+		}
+	};
+
+	return (
+		<Stack
+			gap={0.5}
+			css={{
+				position: 'relative',
+				width: '100%',
```

Comment:
Is this needed if `Box` is `block`? Same on line 217?

Reply:
Well, `Stack` here but `Box` below.

Reply:
The second one may not be necessary, but the first helps prevent layout issues caused by wrappers. For instance, if you wrapped your `TableWrapper` in a `<Stack alignItems="flex-start" justifyContent="flex-start">`, then it won't actually be constrained